cmd/kubelet/app/server.go

```diff
@@ -361,15 +353,37 @@ func (s *KubeletServer) createAPIServerClient() (*client.Client, error) {
 	if len(s.APIServerList) > 1 {
 		glog.Infof("Multiple api servers specified.  Picking first one")
 	}
-	clientConfig.Host = s.APIServerList[0]
 
-	s.addChaosToClientConfig(&clientConfig)
+	// This creates a client, first loading any specified kubeconfig
+	// file, and then overriding the Master flag, if non-empty.
+	clientConfig, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```

Comment:
It's a little weird that explicitly setting `--auth-path` will still result in the `--kubeconfig` default being used instead.   Can you either check the `pflag.Flag.Changed` property or use a `util.StringFlag.Provideded()` to figure out what's been set?

Setting both would be an error, setting one will preferentially use it, setting none will check both default just like this.